app/controllers/passwords_controller.rb

```diff
@@ -30,9 +32,13 @@ def update
   def mfa_edit
     if mfa_edit_conditions_met?
       render template: "passwords/edit"
+    elsif !session_active?
+      login_failure(t("multifactor_auths.session_expired"))
```

Comment:
Might be better to make this `elsif session_active?` and swap the bodies. That way there isn't a negative condition to reason about.

Reply:
The condition above this tests `session_active?` to be true, so by including this in the `elsif`, we can weed out this login failure (expiry) directly, meanwhile the else captures and invalid entry.`